# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1770855213 -39600
# Parent  74c66f2a7e2369f1c232c59ac282aba37364c91f
Misc: Pref mail.folderpane.expand_delay for expanding folders on hover.

diff --git a/mail/app/profile/all-thunderbird.js b/mail/app/profile/all-thunderbird.js
--- a/mail/app/profile/all-thunderbird.js
+++ b/mail/app/profile/all-thunderbird.js
@@ -325,8 +325,9 @@ pref("mail.spellcheck.inline", true);
 pref("mail.folderpane.sizeUnits", "");
 // Summarize messages count and size of subfolders into a collapsed parent?
 // Allowed values: true/false
 pref("mail.folderpane.sumSubfolders", true);
+pref("mail.folderpane.expand_delay", 750);
 
 // target folder URI used for the last move or copy
 pref("mail.last_msg_movecopy_target_uri", "");
 // last move or copy operation was a move
diff --git a/mail/base/content/about3Pane.js b/mail/base/content/about3Pane.js
--- a/mail/base/content/about3Pane.js
+++ b/mail/base/content/about3Pane.js
@@ -3214,14 +3214,20 @@ var folderPane = {
     if (!row?.classList.contains("collapsed")) {
       return;
     }
     this._expandRow = row;
-    this._expandTimer = setTimeout(() => {
-      this._autoExpandedRows.push(this._expandRow);
-      folderTree.expandRow(this._expandRow);
-      delete this._expandRow;
-      delete this._expandTimer;
-    }, 1000);
+    this._expandTimer = setTimeout(
+      () => {
+        this._autoExpandedRows.push(this._expandRow);
+        folderTree.expandRow(this._expandRow);
+        delete this._expandRow;
+        delete this._expandTimer;
+      },
+      Math.max(
+        300,
+        Services.prefs.getIntPref("mail.folderpane.expand_delay", 750)
+      )
+    );
   },
 
   /**
    * Set a timer to collapse all auto-expanded rows in 1000ms.
